HPCC-25482 Split buildall.sh script so can be reused for LN too

jakesmith opened the pull request:
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

jakesmith committed:
HPCC-25482 Split buildall.sh script so can be reused for LN too

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-14668, label: hpcc-25482-buildall-common is skipped.
The reason is: It is related to containerised environment only.
Commit ID: ab04e0eea1dc8996a936b09a22af0f4863e5e79b
OS: centos 7.7.1908 (Linux 3.10.0-327.10.1.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: hpcc-platform-dev-el7-smoketest-2

HPCCSmoketest commented:
Process of PR-14668, label: hpcc-25482-buildall-common is skipped.
The reason is: It is related to containerised environment only.
Commit ID: 7778f5b0fb651d2c68be2bc30a4181d17296ca59
OS: centos 7.7.1908 (Linux 3.10.0-327.10.1.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: hpcc-platform-dev-el7-smoketest-2

jakesmith commented:
@richardkchapman - please review.

Not tested this in context of the github action that calls this..
is there a good way to do that?

jakesmith committed:
Minor changes spotted in review+

1) rename INPUT_BUILDTYPE to INPUT_BUILD_TYPE for consistency
(and reflect in action.yml)

2) move ARG BUILD_TAG back to where it was in platform-build/Dockerfile

3) allow INPUT_BUILD_LABEL to override deduced docker label for
custom docker image versions

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

jakesmith commented on dockerfiles/platform-build/Dockerfile:
```diff
@@ -34,12 +34,12 @@ USER hpcc
 
 ARG BUILD_USER=hpcc-systems
 RUN echo BUILD_USER is ${BUILD_USER}
+ARG BUILD_TAG=none
```
I know why I moved it, so that it would force the clone to be redone if BUILD_TAG changed,
but I don't remember why that was required/desirable..

A cache-layer with an old clone should be ok, as long a it re-fetches, which it would have done if BUILD_TAG changed before..

So I can't see now why it's necessary, happy to move it back.

jakesmith replied:
changed

Michael-Gardner reviewed (dismissed):
I was able to test this both locally and in github actions (with some minor tweaks to my personal repo's master branch). And the LN docker builds worked fine with it.

HPCCSmoketest commented:
Process of PR-14668, label: hpcc-25482-buildall-common is skipped.
The reason is: It is related to containerised environment only.
Commit ID: 017112142312fbf2a8453edb57d6b8fb5bc983a6
OS: centos 7.7.1908 (Linux 3.10.0-327.10.1.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: hpcc-platform-dev-el7-smoketest-2

Michael-Gardner commented on dockerfiles/buildall-common.sh:
```diff
@@ -0,0 +1,161 @@
+[[ -n ${INPUT_BUILD_VER} ]] && BUILD_TAG=${INPUT_BUILD_VER}
+[[ -n ${INPUT_DOCKER_REPO} ]] && DOCKER_REPO=${INPUT_DOCKER_REPO}
+
+if [[ -n ${INPUT_BUILD_THREADS} ]] ; then
```
I don't see a corresponding entry in actions.yml for this variable. Is that going to be an issue if we ever try to pass it in through the github action?

jakesmith replied:
not new (was in buildall.sh) before, but you're right it's not in actions.yml.
I'm not sure we want one though, it defaults to number of detected cores, which is what you prob. want in an action.
I added this (some time ago) to override number concurrent threads used in make locally when building, not to overload my machine.

Michael-Gardner reviewed (approved):
Other than one possible change, looks good.
